dex/src/main/scala/com/wavesplatform/dex/api/http/routes/v0/MarketsRoute.scala

```diff
@@ -344,25 +345,37 @@ final class MarketsRoute(
     (path(AssetPairPM) & delete) { pairOrError =>
       (withMetricsAndTraces("deleteOrderBookWithKey") & protect & withAuth) {
         withAssetPair(assetPairBuilder, pairOrError, validate = false) { pair =>
-          orderBook(pair) match {
-            case Some(Right(_)) =>
-              complete(
-                storeCommand(ValidatedCommand.DeleteOrderBook(pair))
-                  .map {
-                    case None => NotImplemented(error.FeatureDisabled)
-                    case _ => SimpleResponse(StatusCodes.Accepted, "Deleting order book")
-                  }
-                  .recover { case e: Throwable =>
-                    log.error("Can not persist event", e)
-                    CanNotPersist(error.CanNotPersistEvent)
-                  }
-              )
-            case _ => complete(OrderBookUnavailable(error.OrderBookBroken(pair)))
+          withOnlyBlacklistedAssets(pair) {
+            orderBook(pair) match {
+              case Some(Right(_)) =>
+                complete(
+                  storeCommand(ValidatedCommand.DeleteOrderBook(pair))
+                    .map {
+                      case None => NotImplemented(error.FeatureDisabled)
+                      case _ => SimpleResponse(StatusCodes.Accepted, "Deleting order book")
+                    }
+                    .recover { case e: Throwable =>
+                      log.error("Can not persist event", e)
+                      CanNotPersist(error.CanNotPersistEvent)
+                    }
+                )
+              case _ => complete(OrderBookUnavailable(error.OrderBookBroken(pair)))
+            }
           }
         }
       }
     }
 
+  private def withOnlyBlacklistedAssets(assetPair: AssetPair): Directive0 = FutureDirectives.onSuccess(
```

Comment:
An asset pair could be blocked by several approaches. Let's:
1. Rename this to `withOnlyBlacklistedAssetPair`
2. Use `assetPairBuilder.validateAssetPair` instead of checking each asset separately. `AssetPairIsDenied`, `AssetBlacklisted`, `AmountAssetBlacklisted` and `PriceAssetBlacklisted` are considered as success. You can create a trait `error.Blacklisted` and mix it to these errors to write a clean code here.